lib/validation-types/varlist.ts

```diff
@@ -0,0 +1,64 @@
+import * as t from 'io-ts';
+import * as _ from 'lodash';
+
+type AcceptedVarList = VarList | string[];
+
+export interface VarList {
+	[key: string]: string;
+}
+
+const stringRegex = /([^\s=]+?)=(\S+)/;
+
+const validate = (value: unknown): value is VarList => {
+	if (_.isArray(value)) {
+		return validateStringArray(value);
+	} else if (_.isObject(value)) {
+		return _.every(value as { [key: string]: unknown }, (v, k) => {
+			return _.isString(v) && _.isString(k);
+		});
+	}
+	return false;
+};
+
+const convert = (value: unknown): VarList | undefined => {
+	if (!validate(value)) {
+		return;
+	}
+	if (_.isArray(value)) {
+		const varList: VarList = {};
+		_.each(value as string[], str => {
+			const match = str.match(stringRegex);
```

Comment:
why not `if (match === null) return;` ?

Reply:
it's just because due to the validate function above that we know it already exists, but I guess defending against that function changing makes sense, will change.